src/lib/journey/_utilities/metadata.utilities.ts

```diff
@@ -94,6 +98,7 @@ export function buildStepMetadata<T = unknown>(
       numOfSelfSubmittableCbs: callbackMetadataArray.filter((cb) => !!cb.derived.isSelfSubmitting)
         .length,
       numOfUserInputCbs: numOfUserInputCbs,
+      forceButton: shouldForceButton ?? false,
```

Comment:
This is one thing that gave me pause. This seems very UI-centric to be in a pure data analysis portion of this app. Keeping metadata purely about the data is an important aspect to these functions. They should just describe the data, and the UI should then interpret the data to inform how it should render.